### PR TITLE
Fix some doc nits and re-do parts of the Geo IP docs

### PR DESCRIPTION
**Pull request by pieterlexis**
### Short description
cc @cmouse for the geoip doc changes.

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [x] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: docs/backends/geoip.rst [hunk 1]
```diff
@@ -51,30 +51,34 @@ defaults suite you.
 ~~~~~~~~~~~~~~~~~~~~~~~~
 
 .. versionchanged:: 4.2.0
-  This option has been changed since v4.2.0
+  The syntax of the argument has been changed.
```
Comment (cmouse) [inline docs/backends/geoip.rst]: should there be new line before this?

File: docs/backends/geoip.rst [hunk 2]
```diff
@@ -142,95 +146,129 @@ that the ‐ before certain keys is part of the syntax.
...
+.. versionchanged:: 4.2.0
+
+  Before, a record expanded to an empty value it could cause SERVFAIL.
+  Since 4.2.0 such expansions for non-TXT record types are not included in response.
+
+Caching and the GeoIP Backend
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The :ref:`packet-cache` and :ref:`query-cache` will cache the records with EDNS Client Subnet information, when provided in the response.
+Use of certain placeholders (described above) can disable record caching for certail resource records.
```
Comment (cmouse) [inline docs/backends/geoip.rst]: this should probably mention that narrowing scope when serving more-specifics can have sharp corners.

File: docs/backends/geoip.rst (hunk 1 above)
Comment (pieterlexis) [reply docs/backends/geoip.rst]: nope, this is fine :)

File: docs/backends/geoip.rst (hunk 2 above)
Comment (pieterlexis) [reply docs/backends/geoip.rst]: Can you suggest some sentences on how to describe this?

File: docs/backends/geoip.rst [hunk 3]
```diff
@@ -142,95 +146,129 @@ that the ‐ before certain keys is part of the syntax.
...
 
-then caching will not happen for any records of something.example.com.
-If you need to use TXT for debugging, make sure you use dedicated name
-for it.
+  :domain: Defines a domain. You need ttl, records, services under this.
+  :ttl: TTL value for all records, if no TTL is specified in specific record.
+  :records: Records for this domain.
+            Each subkey must be a fully qualified name, under which an array of records follows.
+            Every record is then keyed by its type (e.g. ``a``, ``txt``) and a type may exist more than once.
+            The content for this record may then be confgured as the value.
```
Comment (rgacogne) [inline docs/backends/geoip.rst]: s/confgured/configured/

File: docs/backends/geoip.rst [hunk 4]
```diff
@@ -142,95 +146,129 @@ that the ‐ before certain keys is part of the syntax.
...
+:%wd: Weekday
+:%dd: Year day
+:%hh: Hour
+:%ip: Client IP address
+:%ip4: Client IPv4 address
+:%ip6: Client IPv6 address
+
+.. versionadded:: 4.2.0
+
+  These placeholder have been added in version 4.2.0:
```
Comment (rgacogne) [inline docs/backends/geoip.rst]: s/placeholder/placeholders/

File: docs/backends/geoip.rst [hunk 5]
```diff
@@ -142,95 +146,129 @@ that the ‐ before certain keys is part of the syntax.
...
+
+  - %lat, %lon, %loc to expand for geographic location, if available in backend. %loc in particular can be safely used with LOC record type.
+  - %ip4 and %ip6 that will expand to the IP address when AFI matches, and empty otherwise. Can be particularly used with A and AAAA record types.
+
+.. versionadded:: 4.1.0
+
+  These placeholder have been added in version 4.1.0:
```
Comment (rgacogne) [inline docs/backends/geoip.rst]: s/placeholder/placeholders/

File: docs/backends/geoip.rst [hunk 6]
```diff
@@ -142,95 +146,129 @@ that the ‐ before certain keys is part of the syntax.
...
+Probability is calculated by summing up the weights and dividing each weight with the sum.
+
+Responses to queries
+~~~~~~~~~~~~~~~~~~~~
+
+If the record which a service points to exists under "records" then it is returned as a direct answer.
+If it does not exist under "records" then it is returned as a CNAME.
+
+You can mix service and static records to produce the sum of these records, including apex record.
+For instance, this configuration will send the correct response both A and SOA queries:
```
Comment (rgacogne) [inline docs/backends/geoip.rst]: > this configuration will send the correct response both A and SOA queries

It looks like we might be missing a word?

File: docs/backends/geoip.rst [hunk 7]
```diff
@@ -142,95 +146,129 @@ that the ‐ before certain keys is part of the syntax.
...
+  - services:
+    geo.example.com: ['%co.%cn.example.com']
+
+If your services match wildcard records in your zone file then these will be returned as CNAMEs.
+This will only be an issue if you are trying to use a service record at the apex of your domain where you need other record types to be present (such as NS and SOA records).
+Per :rfc:`2181`, CNAME records cannot appear in the same label as NS or SOA records.
+
+.. versionchanged:: 4.2.0
+
+  Before, a record expanded to an empty value it could cause SERVFAIL.
```
Comment (rgacogne) [inline docs/backends/geoip.rst]: I think we might be missing a word here as well.

File: docs/backends/geoip.rst (hunk 2 above)
Comment (rgacogne) [reply docs/backends/geoip.rst]: s/for certail resource/for certain resource/

File: docs/backends/geoip.rst [hunk 8]
```diff
@@ -142,95 +146,129 @@ that the ‐ before certain keys is part of the syntax.
...
+
+.. code-block:: yaml
+
+  something.example.com:
+    - a: 1.2.3.4
+    - txt: "your ip is %ip"
+
+then caching will not happen for any records of something.example.com.
+
+If you need to use TXT for debugging, make sure you use dedicated name for it.
```
Comment (rgacogne) [inline docs/backends/geoip.rst]: Missing a word as well, perhaps "make sure you use a_a dedicated name for it"?

**Pull request merged**